src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -64,32 +139,63 @@ def create_mhs_device(data, cache) -> Device:
     return product.create_device(**device_payload)
 
 
-def create_party_key_tag(data, cache):
+def create_party_key_tag(data, cache) -> DeviceTagAddedEvent:
     mhs_device: Device = data[create_mhs_device]
-    mhs_device.add_tag(party_key=data[get_party_key])
+    return mhs_device.add_tag(party_key=data[get_party_key])
+
+
+def create_interaction_id_keys(data, cache) -> DeviceKeyAddedEvent:
+    mhs_device: Device = data[create_mhs_device]
+    party_key = data[get_party_key]
+    drd: DeviceReferenceData = data[read_device_reference_data]
+    interaction_ids = []
+
+    # Extract Interaction IDs from questionnaire responses
+    questionnaire_responses = drd.questionnaire_responses.get(
+        "spine_mhs_message_sets/1", []
+    )
+    for response in questionnaire_responses:
+        interaction_ids.append(response.data.get("Interaction ID"))
```

Comment:
I think probably better to pull the field `"Interaction ID"` out of the json schema as we're very likely to rename these fields

Reply:
actually, now I'm thinking (I'm also using this constant) it would be best to have it as a constant, and have a test that checks that this constant matches the json schema